File: JavaKamp/JavaOdevler/4.GunOdev3/Concretes/GameManager.java
package Concrete;

import java.util.ArrayList;
import java.util.List;

import Abstract.GameService;
import Entities.Game;

public class GameManager implements GameService {
	
	List<Game> games = new ArrayList<Game>();

	@Override
	public void add(Game game) {
		this.games.add(game);
	}

	@Override
	public void update(Game game) {
		boolean dataGeldiMi=false;
		int index = 0;
		for (int i=0; i<this.games.size();i++) {
			Game game1 = this.games.get(i);
			if (game.getId()==game1.getId()) {
				index = i;
				dataGeldiMi=true;
				break;
			}
		}
		
		if (dataGeldiMi) {
			this.games.set(index, game);
			System.out.println(game.getName() + " oyun güncellendi!");
		}else {			
			System.err.println("Data bulunamadý!");
		}

	}

	@Override
	public void delete(Game game) {
		this.games.remove(game);

	}

	@Override
	public void getAll() {
		for (Game game : games) {
			System.out.println("Id : " + game.getId() + " Adý: " + game.getName() + " Fiyatý: " + game.getPrice());
		}

	}

}

File: DotNetKamp/GameStoreApplication/GameStoreApplication/Concrete/GameManager.cs
using System;
using System.Collections.Generic;
using System.Text;
using GameStoreApplication.Abstract;
using GameStoreApplication.Entities;

namespace GameStoreApplication
{
    public class GameManager:IGameService
    {
        public List<Game> Games = new List<Game>();
        public void Save(Game game)
        {
            Games.Add(game);
        }

        public void Update(Game game)
        {
            foreach (var gms in Games)
            {
                if (gms == game)
                {
                    int index = Games.IndexOf(gms);
                    Console.WriteLine("{0} indexteki {1} id {2} adında ki oyunun, stok adeti: {3} fiyatı: {4} kayıt bulundu.", index,
                        gms.Id, gms.GameName, gms.Stock, gms.Price);
                    Console.Write("Neyi güncellemek istiyorsunuz? 1- Oyun Adı 2-Stok 3-Fiyat");
                    var secim = Console.ReadLine();
                    int secimInt = Convert.ToInt32(secim);
                    if (secimInt == 1)
                    {
                        Console.Write("Yeni adı giriniz = ");
                        var updateGameName = Console.ReadLine();
                        Games[index].GameName = updateGameName;
                        
                    }
                    else if (secimInt == 2)
                    {
                        Console.Write("Yeni Stok giriniz = ");
                        var updateStock = Console.ReadLine();
                        Games[index].Stock = Convert.ToInt32(updateStock);
                        
                    }
                    else if (secimInt == 3)
                    {
                        Console.Write("Yeni Fiyat giriniz = ");
                        var updatePrice = Console.ReadLine();
                        Games[index].Price = Convert.ToDouble(updatePrice);
                        
                    }
                    else
                    {
                        Console.WriteLine("Hatalı seçim yaptınız!");
                    }
                    Get(gms);
                    break;
                }
                
            }
            Console.WriteLine("Kayıt Bulunamadı!!!!");

        }
        public void Get(Game game)
        {
            Console.WriteLine("{0} id {1} adında {2} stoklu {3} fiyatlı",
                game.Id, game.GameName, game.Stock, game.Price);
        }
        public void Delete(Game game)
        {
            foreach (var gms in Games)
            {
                if (gms == game)
                {
                    int index = Games.IndexOf(gms);
                    Console.WriteLine("{0} indexteki {1} id Kayıt silindi!", index, gms.Id);
                    Games.Remove(gms);
                    break;
                }
                else
                {
                    Console.WriteLine("Kayıt Bulunamadı!!!!");
                }
            }
        }

        public void GetAll()
        {
            Console.WriteLine("----------- Tüm Oyunlar -------------");
            foreach (Game gms in Games)
            {
                Console.WriteLine("{0} id, {1} adındaki oyun, {2} stok adeti, {3} fiyatı",
                    gms.Id, gms.GameName, gms.Stock, gms.Price);
            }
            Console.WriteLine("----------- -------------- -------------");
        }
    }
}
